src/main/java/dev/jbang/source/ResourceRef.java

```diff
@@ -171,6 +171,7 @@ private static ResourceRef forScript(String scriptResource) {
 		try {
 			if (probe != null && probe.canRead()) {
 				if (!probe.getName().endsWith(".jar") && !probe.getName().endsWith(".java")
+						&& !probe.getName().endsWith(".kt")
 						&& !probe.getName().endsWith(".jsh")) {
 					if (probe.isDirectory()) {
 						File defaultApp = new File(probe, "main.java");
```

Comment:
Seeing this line I just remembered that we had a discussion about this. Perhaps we should open an issue to track wether we should remove the support for `main.java` or not...

Reply:
why would we want to remove support for main.java?

Reply:
See, this shows we should turn our talks into issues ASAP, otherwise we keep rehashing things :-)

It was because we wanted to reclaim URLs that point to folders instead of files. Right now I don't remember the exact reason anymore. I'll update when I remember the details :-)

Reply:
What I recall was that we would need to treat urls  with a path ending with / as folder. Main thing was to be able to do jbang https://acme.org/folder/

Reply:
Perhaps. But to have completely different behaviours between http://acme.com and http://acme.com/ is a bit ... awful IMO. Besides the fact that there might be situations where trailing slashes get stripped automatically by the client your're using (officially there should be no semantic difference between `/folder` and `/folder/` after all).